src/encoding/schema/uint64.ts

```diff
@@ -0,0 +1,32 @@
+import { Schema, MsgpackEncodingData, JSONEncodingData } from '../encoding.js';
+import { ensureUint64 } from '../../utils/utils.js';
+
+/* eslint-disable class-methods-use-this */
+
+export class Uint64Schema extends Schema {
+  public defaultValue(): bigint {
+    return BigInt(0);
+  }
+
+  public isDefaultValue(data: unknown): boolean {
```

Comment:
I recall some discussion around the integer type defaults, trying to recall what is impacted by that (not necessarily this function).

Reply:
I think you may be referring to whether integers get decoded as numbers vs bigints?

This function is concerning which values get omitted from the msgpack/JSON encodings. We check for both numbers and bigints to cover all cases.